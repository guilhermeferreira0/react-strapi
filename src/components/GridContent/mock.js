export default {
  title: 'O titulo que eu quero',
  html: `
  <ul>
    <li>Morbi in sem quis dui placerat ornare. Pellentesque odio nisi, euismod in, pharetra a, ultricies in, diam. Sed arcu. Cras consequat.</li>
    <li>Praesent dapibus, neque id cursus faucibus, tortor neque egestas augue, eu vulputate magna eros eu erat. Aliquam erat volutpat. Nam dui mi, tincidunt quis, accumsan porttitor, facilisis luctus, metus.</li>
    <li>Phasellus ultrices nulla quis nibh. Quisque a lectus. Donec consectetuer ligula vulputate sem tristique cursus. Nam nulla quam, gravida non, commodo a, sodales sit amet, nisi.</li>
    <li>Pellentesque fermentum dolor. Aliquam quam lectus, facilisis auctor, ultrices ut, elementum vulputate, nunc.</li>
  </ul>

  <form action="#" method="post">
    <div>
        <label for="name">Text Input:</label>
        <input type="text" name="name" id="name" value="" tabindex="1" />
    </div>

    <div>
        <h4>Radio Button Choice</h4>

        <label for="radio-choice-1">Choice 1</label>
        <input type="radio" name="radio-choice-1" id="radio-choice-1" tabindex="2" value="choice-1" />

      <label for="radio-choice-2">Choice 2</label>
        <input type="radio" name="radio-choice-2" id="radio-choice-2" tabindex="3" value="choice-2" />
    </div>

    <div>
    <label for="select-choice">Select Dropdown Choice:</label>
    <select name="select-choice" id="select-choice">
      <option value="Choice 1">Choice 1</option>
      <option value="Choice 2">Choice 2</option>
      <option value="Choice 3">Choice 3</option>
    </select>
    </div>

    <div>
    <label for="textarea">Textarea:</label>
    <textarea cols="40" rows="8" name="textarea" id="textarea"></textarea>
    </div>

    <div>
      <label for="checkbox">Checkbox:</label>
    <input type="checkbox" name="checkbox" id="checkbox" />
    </div>

    <div>
      <input type="submit" value="Submit" />
    </div>
  </form>`,
  background: false,
};
