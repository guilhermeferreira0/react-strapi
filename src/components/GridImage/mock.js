export default {
  title: 'My grid image',
  description: 'lorem ipsum dolor',
  grid: [
    {
      altText: 'Teste 1',
      srcImage:
        'https://thypix.com/wp-content/uploads/2018/05/Sommerlandschaft-Bilder-80.jpg',
    },
    {
      altText: 'Teste 2',
      srcImage:
        'https://thypix.com/wp-content/uploads/2018/05/Sommerlandschaft-Bilder-80.jpg',
    },
    {
      altText: 'Teste 3',
      srcImage:
        'https://thypix.com/wp-content/uploads/2018/05/Sommerlandschaft-Bilder-80.jpg',
    },
    {
      altText: 'Teste 4',
      srcImage:
        'https://thypix.com/wp-content/uploads/2018/05/Sommerlandschaft-Bilder-80.jpg',
    },
    {
      altText: 'Teste 5',
      srcImage:
        'https://thypix.com/wp-content/uploads/2018/05/Sommerlandschaft-Bilder-80.jpg',
    },
    {
      altText: 'Teste 6',
      srcImage:
        'https://thypix.com/wp-content/uploads/2018/05/Sommerlandschaft-Bilder-80.jpg',
    },
  ],
  background: false,
};
