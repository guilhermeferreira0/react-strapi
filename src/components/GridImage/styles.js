import styled, { css } from 'styled-components';
import { Container as TextComponent } from '../TextComponent/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    & > ${TextComponent} {
      margin-bottom: ${theme.spacings.xhuge};
    }
  `}`;

export const Grid = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacings.large};
  `}`;

export const GridElement = styled.div`
  ${({ theme }) => css`
    overflow: hidden;
    border-radius: 8px;
  `}`;

export const Image = styled.img`
  ${({ theme }) => css`
    width: 100%;
    transition: all .5s ease-in-out;
    cursor: pointer;
    border-radius: 8px;

    &:hover {
      transform: scale(1.4) rotate(10deg);
    }
  `}`;
