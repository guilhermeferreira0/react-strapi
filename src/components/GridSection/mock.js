export default {
  title: 'My grid',
  description: 'lorem ipsum dolor',
  grid: [
    {
      title: 'Teste 1',
      description:
        'lorem ipsum dolor sit amet consecteturadipisicnt esse ut atque doloreis nobis commoix maxime distinctin veritatus und reprehenderet mineus ad dolores privident maiorsds',
    },
    {
      title: 'Teste 2',
      description:
        'lorem ipsum dolor sit amet consecteturadipisicnt esse ut atque doloreis nobis commoix maxime distinctin veritatus und reprehenderet mineus ad dolores privident maiorsds',
    },
    {
      title: 'Teste 3',
      description:
        'lorem ipsum dolor sit amet consecteturadipisicnt esse ut atque doloreis nobis commoix maxime distinctin veritatus und reprehenderet mineus ad dolores privident maiorsds',
    },
  ],
  background: false,
};
