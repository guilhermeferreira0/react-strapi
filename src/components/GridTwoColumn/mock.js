export default {
  title: 'GridTwoColumns',
  text: 'lorem ipsuma sodkaol Ã§lsdp spdlpdlsd sd lrrem lomr lomrem',
  srcImage: 'assets/images/logo.svg',
  background: false,
};
