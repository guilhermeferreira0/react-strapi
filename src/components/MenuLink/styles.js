import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => css`
    display: block;
    text-decoration: none;
    font-size: ${theme.fonts.sizes.small};
    padding: ${theme.spacings.small};
    color: ${theme.colors.primaryColor};
    position: relative;
    scale: 1;
    transition: scale 0.5s ease-in-out;


    &::after {
      content: '';
      position: absolute;
      bottom: 0.75rem;
      left: 50%;
      width: 0%;
      height: 0.2rem;
      background-color: ${theme.colors.secondaryColor};
      transition: all 0.5s ease-in-out;
    }

    &:hover::after {
      left: 25%;
      width: 50%;
    }

    &:hover {
      scale: 1.1;
    }

    @media ${theme.media.ltemedium} {
      &:hover::after {
      left: 20.5%;
      width: 50%;
    }
    }
  `}`;
