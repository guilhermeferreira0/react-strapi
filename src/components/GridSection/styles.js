import styled, { css } from 'styled-components';
import { Title as HeadingContainer } from '../Heading/styles';
import { Container as TextComponent } from '../TextComponent/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    ${TextComponent} {
      margin-bottom: ${theme.spacings.xhuge};
    }
  `}`;

export const Grid = styled.div`
  ${({ theme }) => css`
    counter-reset: grid-counter;
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacings.large};

    @media ${theme.media.ltemedium} {
      grid-template-columns: 1fr;
    }
  `}`;

export const GridElement = styled.div`
  ${({ theme }) => css`
    ${HeadingContainer} {
      position: relative;
      left: 7rem;
    }

    ${HeadingContainer}::before {
      content: counter(grid-counter);
      counter-increment: grid-counter;
      position: absolute;
      font-size: ${theme.fonts.sizes.huge};
      top: -2rem;
      left: -7rem;
      transform: rotate(10deg);
    }
  `}`;
