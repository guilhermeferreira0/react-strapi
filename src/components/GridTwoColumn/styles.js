import styled, { css } from 'styled-components';
import { Title } from '../Heading/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: 1fr 2fr;
    align-items: center;
    gap: ${theme.spacings.large};

    @media ${theme.media.ltemedium} {
      grid-template-columns: 1fr;
      text-align: center;
    }

    ${Title} {
      margin-bottom: ${theme.spacings.xlarge};
    }
  `}`;

export const TextContainer = styled.div`
  ${({ theme }) => css`
    @media ${theme.media.ltemedium} {
      margin-bottom: ${theme.spacings.large};
    }
  `}`;

export const ImageContainer = styled.div`
  ${({ theme }) => css``}`;

export const Image = styled.div`
  ${({ theme }) => css``}`;
