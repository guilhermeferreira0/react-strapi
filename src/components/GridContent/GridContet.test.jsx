import { screen } from '@testing-library/react';
import { GridContent } from '.';
import { renderTheme } from '../../styles/renderTheme';
import mock from './mock';

describe('<GridContent />', () => {
  it('Should render grid content', () => {
    const { container } = renderTheme(<GridContent {...mock} />);
    expect(container).toMatchSnapshot();
  });

  it('Should render grid content', () => {
    const { container } = renderTheme(
      <GridContent {...mock} background={undefined} />,
    );
    expect(container).toMatchSnapshot();
  });
});
