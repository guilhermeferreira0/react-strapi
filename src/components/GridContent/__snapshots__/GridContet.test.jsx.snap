// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GridContent /> Should render grid content 1`] = `
.c2 {
  text-align: center;
  max-width: 58rem;
  margin: 0 auto;
}

.c4 {
  margin: 6.2rem 0;
}

.c3 {
  color: black;
  font-size: 6.2rem;
  text-transform: uppercase;
}

.c5 {
  font-size: 2.4rem;
}

.c0 {
  background-color: #FFFFFF;
  color: black;
  min-height: 100vh;
  display: flex;
  align-items: center;
}

.c1 {
  max-width: 120rem;
  margin: 0 auto;
  padding: 3.2rem;
}

@media (max-width:768px) {
  .c3 {
    font-size: 4rem;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <h2
          class="c3"
        >
          O titulo que eu quero
        </h2>
        <div
          class="c4"
        >
          <div
            class="c5"
          >
            
  
            <ul>
              
    
              <li>
                Morbi in sem quis dui placerat ornare. Pellentesque odio nisi, euismod in, pharetra a, ultricies in, diam. Sed arcu. Cras consequat.
              </li>
              
    
              <li>
                Praesent dapibus, neque id cursus faucibus, tortor neque egestas augue, eu vulputate magna eros eu erat. Aliquam erat volutpat. Nam dui mi, tincidunt quis, accumsan porttitor, facilisis luctus, metus.
              </li>
              
    
              <li>
                Phasellus ultrices nulla quis nibh. Quisque a lectus. Donec consectetuer ligula vulputate sem tristique cursus. Nam nulla quam, gravida non, commodo a, sodales sit amet, nisi.
              </li>
              
    
              <li>
                Pellentesque fermentum dolor. Aliquam quam lectus, facilisis auctor, ultrices ut, elementum vulputate, nunc.
              </li>
              
  
            </ul>
            

  
            <form
              action="#"
              method="post"
            >
              
    
              <div>
                
        
                <label
                  for="name"
                >
                  Text Input:
                </label>
                
        
                <input
                  id="name"
                  name="name"
                  tabindex="1"
                  type="text"
                  value=""
                />
                
    
              </div>
              

    
              <div>
                
        
                <h4>
                  Radio Button Choice
                </h4>
                

        
                <label
                  for="radio-choice-1"
                >
                  Choice 1
                </label>
                
        
                <input
                  id="radio-choice-1"
                  name="radio-choice-1"
                  tabindex="2"
                  type="radio"
                  value="choice-1"
                />
                

      
                <label
                  for="radio-choice-2"
                >
                  Choice 2
                </label>
                
        
                <input
                  id="radio-choice-2"
                  name="radio-choice-2"
                  tabindex="3"
                  type="radio"
                  value="choice-2"
                />
                
    
              </div>
              

    
              <div>
                
    
                <label
                  for="select-choice"
                >
                  Select Dropdown Choice:
                </label>
                
    
                <select
                  id="select-choice"
                  name="select-choice"
                >
                  
      
                  <option
                    value="Choice 1"
                  >
                    Choice 1
                  </option>
                  
      
                  <option
                    value="Choice 2"
                  >
                    Choice 2
                  </option>
                  
      
                  <option
                    value="Choice 3"
                  >
                    Choice 3
                  </option>
                  
    
                </select>
                
    
              </div>
              

    
              <div>
                
    
                <label
                  for="textarea"
                >
                  Textarea:
                </label>
                
    
                <textarea
                  cols="40"
                  id="textarea"
                  name="textarea"
                  rows="8"
                />
                
    
              </div>
              

    
              <div>
                
      
                <label
                  for="checkbox"
                >
                  Checkbox:
                </label>
                
    
                <input
                  id="checkbox"
                  name="checkbox"
                  type="checkbox"
                />
                
    
              </div>
              

    
              <div>
                
      
                <input
                  type="submit"
                  value="Submit"
                />
                
    
              </div>
              
  
            </form>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<GridContent /> Should render grid content 2`] = `
.c2 {
  text-align: center;
  max-width: 58rem;
  margin: 0 auto;
}

.c4 {
  margin: 6.2rem 0;
}

.c3 {
  color: black;
  font-size: 6.2rem;
  text-transform: uppercase;
}

.c5 {
  font-size: 2.4rem;
}

.c0 {
  background-color: #FFFFFF;
  color: black;
  min-height: 100vh;
  display: flex;
  align-items: center;
}

.c1 {
  max-width: 120rem;
  margin: 0 auto;
  padding: 3.2rem;
}

@media (max-width:768px) {
  .c3 {
    font-size: 4rem;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <h2
          class="c3"
        >
          O titulo que eu quero
        </h2>
        <div
          class="c4"
        >
          <div
            class="c5"
          >
            
  
            <ul>
              
    
              <li>
                Morbi in sem quis dui placerat ornare. Pellentesque odio nisi, euismod in, pharetra a, ultricies in, diam. Sed arcu. Cras consequat.
              </li>
              
    
              <li>
                Praesent dapibus, neque id cursus faucibus, tortor neque egestas augue, eu vulputate magna eros eu erat. Aliquam erat volutpat. Nam dui mi, tincidunt quis, accumsan porttitor, facilisis luctus, metus.
              </li>
              
    
              <li>
                Phasellus ultrices nulla quis nibh. Quisque a lectus. Donec consectetuer ligula vulputate sem tristique cursus. Nam nulla quam, gravida non, commodo a, sodales sit amet, nisi.
              </li>
              
    
              <li>
                Pellentesque fermentum dolor. Aliquam quam lectus, facilisis auctor, ultrices ut, elementum vulputate, nunc.
              </li>
              
  
            </ul>
            

  
            <form
              action="#"
              method="post"
            >
              
    
              <div>
                
        
                <label
                  for="name"
                >
                  Text Input:
                </label>
                
        
                <input
                  id="name"
                  name="name"
                  tabindex="1"
                  type="text"
                  value=""
                />
                
    
              </div>
              

    
              <div>
                
        
                <h4>
                  Radio Button Choice
                </h4>
                

        
                <label
                  for="radio-choice-1"
                >
                  Choice 1
                </label>
                
        
                <input
                  id="radio-choice-1"
                  name="radio-choice-1"
                  tabindex="2"
                  type="radio"
                  value="choice-1"
                />
                

      
                <label
                  for="radio-choice-2"
                >
                  Choice 2
                </label>
                
        
                <input
                  id="radio-choice-2"
                  name="radio-choice-2"
                  tabindex="3"
                  type="radio"
                  value="choice-2"
                />
                
    
              </div>
              

    
              <div>
                
    
                <label
                  for="select-choice"
                >
                  Select Dropdown Choice:
                </label>
                
    
                <select
                  id="select-choice"
                  name="select-choice"
                >
                  
      
                  <option
                    value="Choice 1"
                  >
                    Choice 1
                  </option>
                  
      
                  <option
                    value="Choice 2"
                  >
                    Choice 2
                  </option>
                  
      
                  <option
                    value="Choice 3"
                  >
                    Choice 3
                  </option>
                  
    
                </select>
                
    
              </div>
              

    
              <div>
                
    
                <label
                  for="textarea"
                >
                  Textarea:
                </label>
                
    
                <textarea
                  cols="40"
                  id="textarea"
                  name="textarea"
                  rows="8"
                />
                
    
              </div>
              

    
              <div>
                
      
                <label
                  for="checkbox"
                >
                  Checkbox:
                </label>
                
    
                <input
                  id="checkbox"
                  name="checkbox"
                  type="checkbox"
                />
                
    
              </div>
              

    
              <div>
                
      
                <input
                  type="submit"
                  value="Submit"
                />
                
    
              </div>
              
  
            </form>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
